tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs

```diff
@@ -0,0 +1,157 @@
+    using global::AudioAnalysisTools.DSP;
+    using global::AudioAnalysisTools.EventStatistics;
+    using global::AudioAnalysisTools.WavTools;
+    using global::TowseyLibrary;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using SixLabors.ImageSharp;
+    using SixLabors.ImageSharp.PixelFormats;
+
+    [TestClass]
+    public class EventTests
+    {
+        [TestMethod]
+        public void TestEventMerging()
+        {
+            // make a list of events
+            var events = new List<AcousticEvent>();
+
+            var segmentStartOffset = TimeSpan.Zero;
+            double maxPossibleScore = 10.0;
+            var startTime1 = 1.0;
+            var duration1 = 5.0;
+            var minHz1 = 1000;
+            var maxHz1 = 8000;
+            var event1 = new AcousticEvent(segmentStartOffset, startTime1, duration1, minHz1, maxHz1)
+            {
+                Name = "Event1",
+                Score = 1.0,
+                ScoreNormalised = 1.0 / maxPossibleScore,
+            };
+
+            events.Add(event1);
+
+            var startTime2 = 4.5;
+            var duration2 = 2.0;
+            var minHz2 = 1500;
+            var maxHz2 = 6000;
+            var event2 = new AcousticEvent(segmentStartOffset, startTime2, duration2, minHz2, maxHz2)
+            {
+                Name = "Event2",
+                Score = 5.0,
+                ScoreNormalised = 5.0 / maxPossibleScore,
+            };
+            events.Add(event2);
+
+            var startTime3 = 7.0;
+            var duration3 = 2.0;
+            var minHz3 = 1000;
+            var maxHz3 = 8000;
+            var event3 = new AcousticEvent(segmentStartOffset, startTime3, duration3, minHz3, maxHz3)
+            {
+                Name = "Event3",
+                Score = 9.0,
+                ScoreNormalised = 9.0 / maxPossibleScore,
+            };
+            events.Add(event3);
+
+            // combine adjacent acoustic events
+            events = AcousticEvent.CombineOverlappingEvents(events, segmentStartOffset);
+
+            Assert.AreEqual(2, events.Count);
+            Assert.AreEqual(1.0, events[0].EventStartSeconds, 1E-4);
+            Assert.AreEqual(6.5, events[0].EventEndSeconds, 1E-4);
+            Assert.AreEqual(7.0, events[1].EventStartSeconds, 1E-4);
+            Assert.AreEqual(9.0, events[1].EventEndSeconds, 1E-4);
+
+            Assert.AreEqual(1000, events[0].LowFrequencyHertz);
+            Assert.AreEqual(8000, events[0].HighFrequencyHertz);
+            Assert.AreEqual(1000, events[1].LowFrequencyHertz);
+            Assert.AreEqual(8000, events[1].HighFrequencyHertz);
+
+            Assert.AreEqual(5.0, events[0].Score, 1E-4);
+            Assert.AreEqual(9.0, events[1].Score, 1E-4);
+            Assert.AreEqual(0.5, events[0].ScoreNormalised, 1E-4);
+            Assert.AreEqual(0.9, events[1].ScoreNormalised, 1E-4);
+        }
+
+        [TestMethod]
+        public void TestSonogramWithEventsOverlay()
```

Comment:
For this test, I'd suggest using the `GeneratedImageTest` base class I've set up.

Also, the previous comment about ceremony applies here